src/io/port.rs

```diff
@@ -0,0 +1,94 @@
+//! Solaris/Illumos event ports.
+
+use crate::backend::{c, io::syscalls};
+use crate::fd::{AsFd, AsRawFd, OwnedFd};
+use crate::io;
+
+use super::PollFlags;
+
+use core::time::Duration;
+
+/// The structure representing a port event.
+#[repr(transparent)]
+pub struct Event(pub(crate) c::port_event);
+
+impl Event {
+    /// Get the events associated with this event.
+    pub fn events(&self) -> i32 {
+        self.0.portev_events
+    }
+
+    /// Get the event source associated with this event.
+    pub fn object(&self) -> usize {
+        self.0.portev_object
+    }
+
+    /// Get the userdata associated with this event.
+    pub fn userdata(&self) -> *mut () {
+        self.0.portev_user.cast()
+    }
+}
+
+/// `port_create()`- Creates a new port.
+pub fn port_create() -> io::Result<OwnedFd> {
+    syscalls::port_create()
+}
+
+/// `port_associate()`- Associates a file descriptor with a port.
```

Comment:
The current comment convention is to include the argument names in the pseudocode comments.